Solutions/Marain.UserNotifications.Management.Host/Marain/UserNotifications/Management/Host/Composer/GenerateTemplateComposer.cs

```diff
@@ -41,7 +41,7 @@ public GenerateTemplateComposer(ILogger<GenerateTemplateComposer> logger)
             EmailTemplate? emailTemplate = null;
             SmsTemplate? smsTemplate = null;
             WebPushTemplate? webPushTemplate = null;
-            Dictionary<string, object> existingProperties = PropertyBagHelpers.GetDictionaryFromPropertyBag(body);
+            var existingProperties = body.AsDictionaryRecursive().ToDictionary(x => x.Key, x => x.Value);
```

Comment:
Wait, what? Why do we need to call `ToDictionary` on the value returned by `AsDictionaryRecursive`? I could see it if there was some sort of rewrite but the lambdas here shouldn't change anything.

Reply:
Code further down the stack is expecting an `IDictionary<string, object>`; `IReadOnlyDictionary<string, object>`, as returned from `AsDictionaryRecursive` does not implement that interface.

Reply:
Is the code further down the stack the place that should be doing the conversion to a writable dictionary if that is what it needs at that point?

Reply:
The code certainly looks like it's only reading from the dictionary, so ideally it shouldn't need a writeable one. Where does `Hash.FromDictionary` and is it insisting on a non-readonly one?

Reply:
No. Ultimately it gets fed into `DotLiquid.Hash.FromDictionary` to convert it to the DotLiquid `Hash` form. I've actually pulled this up a level as in its current form, we'd be doing the dictionary -> hash conversion multiple times for no reason.

So, there's scope in `DotLiquid.Hash` to support creating a new instance from an `IEnumerable<KeyValuePair<string, object>>` which would make it more flexible, but that's for a separate PR I think.

Reply:
I've created https://github.com/corvus-dotnet/Corvus.DotLiquidAsync/pull/115 to address this.